### PR TITLE
Fix Executable base path

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Explanation please!

**Comment by sebastienros** [comment]
@BrennanConroy, sorry ;)

There is a bug that when you pass `executable: foo.exe` it will look for `published\foo.exe` because it was adding `\published` in all cases. Now it's not adding it if `executable: ` is set.

The other part is to force a project to exclude a `published` folder from the publication because otherwise it would recursively copy this folder after each build is the repository is cached on the agent. This is a mitigation for another bug (`noBuild: true`, detailed in an open issue) until I find why it not working anymore (we know which commit broke the feature).

File: src/Microsoft.Crank.Agent/Startup.cs [hunk 1]
```diff
@@ -3704,6 +3704,16 @@ private static async Task PatchProjectFrameworkReferenceAsync(Job job, string pr
                         );
                     }
 
+                    // Exclude "published" folder from content to prevent recursively copying it after each cached build
+                    var excludePublishedGroup = new XElement("ItemGroup");
+                    project.Root.Add(excludePublishedGroup);
```
Comment (BrennanConroy) [inline src/Microsoft.Crank.Agent/Startup.cs:3709]: Should this be added after we update the `excludePublishedGroup`? Just in case it copies on add.

Comment (sebastienros) [reply src/Microsoft.Crank.Agent/Startup.cs:3709]: You might be right, I see something like this:

```xml
  <ItemGroup />
  <ItemGroup>
    <Content Update="published\**" CopyToPublishDirectory="Never" />
  </ItemGroup>
```

**Review by BrennanConroy — APPROVED** [review]

**Comment by RobertHenry6bev** [comment]
I rebase to upstream, and am still getting failures

```
  pmu_getter:
    executable: pmu_support/pmu_getter.sh
    readyStateText: STARTED
    isConsoleApp: true

```

```
11:06:39 INF] Skipping build step, no required
[11:06:39 INF] [/mnt/robhenry/crank_0/dotnet_bench/crank/src/Microsoft.Crank.Agent] chmod +x /mnt/robhenry/crank_0/robhenry_crank_0/benchmarks-server-380025/24jo1wej.isl/pmu_support/pmu_getter.sh
[11:06:39 INF] [STDERR] chmod: cannot access '/mnt/robhenry/crank_0/robhenry_crank_0/benchmarks-server-380025/24jo1wej.isl/pmu_support/pmu_getter.sh': No such file or directory
[11:06:40 INF] An unexpected error occurred while building the job. System.AggregateException: One or more errors occurred. (Command chmod +x /mnt/robhenry/crank_0/robhenry_crank_0/benchmarks-server-380025/24jo1wej.isl/pmu_support/pmu_getter.sh returned exit code 1)
 ---> System.InvalidOperationException: Command chmod +x /mnt/robhenry/crank_0/robhenry_crank_0/benchmarks-server-380025/24jo1wej.isl/pmu_support/pmu_getter.sh returned exit code 1
   at Microsoft.Crank.Agent.ProcessUtil.RunAsync(String filename, String arguments, Nullable`1 timeout, String workingDirectory, Boolean throwOnError, IDictionary`2 environmentVariables, Action`1 outputDataReceived, Boolean log, Action`1 onStart, Action`1 onStop, Boolean captureOutput, Boolean captureError, CancellationToken cancellationToken) in /mnt/robhenry/crank_0/dotnet_bench/crank/src/Microsoft.Crank.Agent/ProcessUtil.cs:line 166
   at Microsoft.Crank.Agent.ProcessUtil.RunAsync(String filename, String arguments, Nullable`1 timeout, String workingDirectory, Boolean throwOnError, IDictionary`2 environmentVariables, Action`1 outputDataReceived, Boolean log, Action`1 onStart, Action`1 onStop, Boolean captureOutput, Boolean captureError, CancellationToken cancellationToken) in /mnt/robhenry/crank_0/dotnet_bench/crank/src/Microsoft.Crank.Agent/ProcessUtil.cs:line 168
   at Microsoft.Crank.Agent.Startup.StartProcess(String hostname, String benchmarksRepo, Job job, String dotnetHome, JobContext context) in /mnt/robhenry/crank_0/dotnet_bench/crank/src/Microsoft.Crank.Agent/Startup.cs:line 4563
   at Microsoft.Crank.Agent.Startup.<>c__DisplayClass81_4.<<ProcessJobs>b__22>d.MoveNext() in /mnt/robhenry/crank_0/dotnet_bench/crank/src/Microsoft.Crank.Agent/Startup.cs:line 909
   --- End of inner exception stack trace ---
[11:06:40 INF] Starting -> Failed (pmu_getter:10)
```

**Pull request merged**